plugins/modules/junos_routing_options.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for junos_routing_options
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: junos_routing_options
+version_added: 2.8.0
+short_description: Manage routing-options configuration on Junos devices.
+description: This module manages routing-options configuration on devices running Junos.
+author: Rohit Thakur (@rohitthakur2590)
+requirements:
+  - ncclient (>=v0.6.4)
+  - xmltodict (>=0.12.0)
+notes:
+  - This module requires the netconf system service be enabled on the device being managed.
+  - This module works with connection C(netconf).
+  - See L(the Junos OS Platform Options,https://docs.ansible.com/ansible/latest/network/user_guide/platform_junos.html).
+  - Tested against JunOS v18.4R1
+options:
+  running_config:
+    description:
+    - This option is used only with state I(parsed).
+    - The value of this option should be the output received from the Junos device
+      by executing the command B(show system routing-options).
+    - The state I(parsed) reads the configuration from C(running_config) option and
+      transforms it into Ansible structured data as per the resource module's argspec
+      and the value is then returned in the I(parsed) key within the result.
+    type: str
+  config:
+    description: A dictionary of routing-options configuration.
+    type: dict
+    suboptions:
+      autonomous_system:
+        description: Specify Autonomous system number.
+        type: dict
+        suboptions:
+          as_number:
```

Comment:
If let's say BGP is already configured and running on the device, what happens if you remove the as_number using this module? Should we allow this or should be introspect the running-config and block the deletion?

Reply:
First of all, that's a really good question, so as for Junos even BGP module separately fetches this from routing options and manages AS config(as we decided back then), So with this module, it will simply remove the autonomous-system config with the deleted operation. We need to document this in BGP modules and routing-options RM.
If we think about going toward introspection or even having something like purged support, IMHO we would be overcomplicating things here.